ui/app/js/services/rules_service.js

```diff
@@ -88,6 +88,49 @@ angular.module("app").factory('Rules', function($http, ScheduledChanges, Helpers
       url += "?csrf_token=" + encodeURIComponent(data["csrf_token"]);
       return $http.delete(url, data);
     },
+    ruleSignoffsRequired: function(oldRule, newRule, productSignoffRequirements) {
```

Comment:
What do you think about having the backend return this instead of calculating it here? We already do it for the Scheduled Changes endpoints (eg: https://github.com/mozilla/balrog/blob/master/auslib/web/admin/views/scheduled_changes.py#L59), and it would allow you to re-use the underlying database layer code that already knows how to do this.

Reply:
How would that work? We would send a request on every change? It doesn't seem like a big win to me because the backend code is focused on database entries, whereas a potential change isn't in the database yet.

Reply:
Hmmm, I think I got my use cases mixed up. I was thinking about https://bugzilla.mozilla.org/show_bug.cgi?id=1343915, which requires information about existing objects. I agree that for this use case (signoff requirements for stuff that doesn't exist yet), this makes sense. Sorry!